### PR TITLE
exploit fixes (untested)

### PR DESCRIPTION
**Pull request by Xytriza1**
not tested. used snyk x copilot so there may be some incorrect fixes.

removed option for tool custom url & debug as it can be sometimes a security risk

### CODE REVIEW

**Comment by Xytriza1** [comment]
reply if something needs to be changed & also would like to know if testing goes well

File: incl/misc/getAccountURL.php [hunk 1]
```diff
@@ -1,3 +1,6 @@
 <?php
-$url = (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+$host = htmlspecialchars($_SERVER['HTTP_HOST'], ENT_QUOTES, 'UTF-8');
+$request_uri = htmlspecialchars($_SERVER['REQUEST_URI'], ENT_QUOTES, 'UTF-8');
+
+$url = (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$host$request_uri";
```
Comment (Cvolton) [inline incl/misc/getAccountURL.php:5]: this is almost certainly gonna break stuff

File: incl/mods/requestUserAccess.php [hunk 2]
```diff
@@ -13,6 +13,6 @@
 	if ($permState >= 2){ // if the mod badge level is higher than 2, it will still show elder mod message
 		exit("2");
 	}
-	echo $permState; 
+	echo htmlspecialchars($permState, ENT_QUOTES, 'UTF-8');
```
Comment (Cvolton) [inline incl/mods/requestUserAccess.php:16]: most of these `htmlspecialchars` calls feel pointless, as the value is sanitized elsewhere

File: accounts/backupGJAccount.php [hunk 3]
```diff
@@ -61,8 +61,9 @@
 	}*/
 	//$query = $db->prepare("UPDATE `accounts` SET `saveData` = :saveData WHERE userName = :userName");
 	//$query->execute([':saveData' => $saveData, ':userName' => $userName]);
-	file_put_contents("../data/accounts/$accountID",$saveData);
-	file_put_contents("../data/accounts/keys/$accountID","");
+	$safeAccountID = basename($accountID);
```
Comment (Cvolton) [inline accounts/backupGJAccount.php:64]: this doesn't hurt but it feels pointless when the value is guaranteed to be an int

**Review by Cvolton — COMMENTED** [review]
need to take a more detailed look later, the htmlspecialchars in gd endpoints feel incredibly pointless tho

**Comment by Xytriza1** [comment]
i mean true but it was a xss fix so i'm not sure

**Comment by MegaSa1nt** [comment]
> need to take a more detailed look later, the htmlspecialchars in gd endpoints feel incredibly pointless tho

+ there is cool dashboard with many security improvements and fixes :trollface:

**Comment by Xytriza1** [comment]
wdym? i didn't write any code it was all just snyk and copilot lmfao

**Pull request closed without merging**